src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java

```diff
@@ -0,0 +1,855 @@
+     * correlate to this value!</b>
+     *
+     * @return time in milliseconds between heartbeat and the heartbeat ack response
+     */
+    default double getAveragePing()
+    {
+        return this.getShardCache()
+                .stream()
+                .mapToLong(jda -> jda.getPing())
+                .filter(ping -> ping != -1)
+                .average()
+                .getAsDouble();
+    }
+
+    /**
+     * Gets all {@link net.dv8tion.jda.core.entities.Category Categories} visible to the currently logged in account.
+     *
+     * @return An immutable list of all visible {@link net.dv8tion.jda.core.entities.Category Categories}.
+     */
+    default List<Category> getCategories()
+    {
+        return this.getCategoryCache().asList();
+    }
+
+    /**
+     * Gets a list of all {@link net.dv8tion.jda.core.entities.Category Categories} that have the same name as the one
+     * provided. <br>If there are no matching categories this will return an empty list.
+     *
+     * @param name
+     *         The name to check
+     * @param ignoreCase
+     *         Whether to ignore case on name checking
+     * @return Immutable list of all categories matching the provided name
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided name is {@code null}
+     */
+    default List<Category> getCategoriesByName(final String name, final boolean ignoreCase)
+    {
+        return this.getCategoryCache().getElementsByName(name, ignoreCase);
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.entities.Category Category} that matches the provided id. <br>If there is no
+     * matching {@link net.dv8tion.jda.core.entities.Category Category} this returns {@code null}.
+     *
+     * @param id
+     *         The snowflake ID of the wanted Category
+     * @return Possibly-null {@link net.dv8tion.jda.core.entities.Category Category} for the provided ID.
+     */
+    default Category getCategoryById(final long id)
+    {
+        return this.getCategoryCache().getElementById(id);
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.entities.Category Category} that matches the provided id. <br>If there is no
+     * matching {@link net.dv8tion.jda.core.entities.Category Category} this returns {@code null}.
+     *
+     * @param id
+     *         The snowflake ID of the wanted Category
+     * @return Possibly-null {@link net.dv8tion.jda.core.entities.Category Category} for the provided ID.
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided ID is not a valid {@code long}
+     */
+    default Category getCategoryById(final String id)
+    {
+        return this.getCategoryCache().getElementById(id);
+    }
+
+    /**
+     * {@link net.dv8tion.jda.core.utils.cache.SnowflakeCacheView SnowflakeCacheView} of
+     * all cached {@link net.dv8tion.jda.core.entities.Category Categories} visible to this ShardManager session.
```

Comment:
Its not a "session" but an instance. Sessions is what we call actual discord connections.